ext/node/polyfills/internal_binding/stream_wrap.ts

```diff
@@ -331,7 +336,17 @@ export class LibuvStreamWrap extends HandleWrap {
     }
 
     try {
-      nread = await this[kStreamBaseField]!.read(buf);
+      if (this[kStreamBaseField]![_readCancel]) {
+        const { cancelHandle, nread: p } = this[kStreamBaseField]!
+          [_readCancel](buf);
+        if (cancelHandle) {
+          this.cancelHandle = cancelHandle;
+        }
+
+        nread = await p;
+      } else {
+        nread = await this[kStreamBaseField]!.read(buf);
```

Comment:
Ugh, this file should probably be written in JS, I don't think TS helps with anything here...